src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/domainEvents/OutboundEventsPublisher.kt

```diff
@@ -26,20 +32,28 @@ class OutboundEventsPublisher(
   }
   private val domainEventsTopicClient by lazy { domainEventsTopic.snsClient }
 
-  fun publishDomainEvent(event: HMPPSDomainEvent, licenceId: String) {
+  @TransactionalEventListener(phase = TransactionPhase.AFTER_COMMIT)
+  fun publishDomainEvent(event: HMPPSDomainEvent) {
     val eventType = event.eventType
-    log.debug("Event {} for licence ID {}", eventType, licenceId)
-    domainEventsTopicClient.publish(
+    val publishedDomainEvent = buildDomainEvent(event).get(FUTURE_TIMEOUT, TimeUnit.MILLISECONDS)
+    if (publishedDomainEvent != null) {
```

Comment:
Rather than `buildDomainEvent` this is raising the event.  `publishedDomainEvent` is really the `publishResponse`

I don't think it can ever be null. 
* If the future hasn't completed then a timeout exception will be thrown
* If the future completes with an exception, then an ExecutionException will be thrown with the exception as its cause
* If the future completes successfully the object returned will be the response.